From 5660e365610394727c0ee8988231f3dd544b6903 Mon Sep 17 00:00:00 2001
From: ran shalit <rshalit@habana.ai>
Date: Tue, 16 Nov 2021 14:45:20 +0200
Subject: [PATCH] [SW-62501] boot_info: use specs boot_info struct

Use boot_info struct from spec, instead of defining
the same struct in each project.

Change-Id: Ifd755cde5d6d407db296b3ca5a32c02737df538b
Signed-off-by: ran shalit <rshalit@habana.ai>
---

diff --git a/lib/preboot/boot_info.c b/lib/preboot/boot_info.c
index ab1c396..aa4c59b 100644
--- a/lib/preboot/boot_info.c
+++ b/lib/preboot/boot_info.c
@@ -15,22 +15,7 @@
 #include "boot_info.h"
 #include "uart.h"
 
-#define _MAX_INFO_DESCS	32
-
-#define _BOOT_INFO_SZ	(sizeof(struct hl_boot_info_ptrs) + \
-			sizeof(struct hl_boot_info_desc) * _MAX_INFO_DESCS)
-
-struct _boot_info {
-	union {
-		struct {
-			struct hl_boot_info_ptrs ptrs;
-			struct hl_boot_info_desc descs[_MAX_INFO_DESCS];
-		};
-		u32 data[ROUND_UP4(_BOOT_INFO_SZ) / sizeof(u32)];
-	};
-};
-
-static struct _boot_info boot_info = { .ptrs = { .count = 0 } };
+static struct boot_info boot_info = { .ptrs = { .count = 0 } };
 
 static inline void *offset2ptr(u32 offset, uintptr_t base)
 {
@@ -88,12 +73,12 @@
 void boot_info_finalize(void)
 {
 	struct hl_boot_info_desc *desc;
-	u32 offsets[_MAX_INFO_DESCS];
+	u32 offsets[MAX_INFO_DESCS];
 	u32 offset, copy_sz;
 	const u32 *src;
 	u32 *dst;
 	int i, j;
-	struct hl_boot_info_desc descs[_MAX_INFO_DESCS];
+	struct hl_boot_info_desc descs[MAX_INFO_DESCS];
 
 	if (!boot_info.ptrs.count)
 		return;
@@ -157,7 +142,7 @@
 
 	desc = boot_info_get_desc(type);
 	if (!desc) {
-		if (boot_info.ptrs.count >= _MAX_INFO_DESCS) {
+		if (boot_info.ptrs.count >= MAX_INFO_DESCS) {
 			preboot_puts_hex("reached max desc count ", type, 0);
 			return 1;
 		}
